src/Connections/FeedConnection.ts

```diff
@@ -0,0 +1,68 @@
+import {Appservice} from "matrix-bot-sdk";
+import { IConnection, IConnectionState } from ".";
+import { BridgeConfigFeeds } from "../Config/Config";
+import { FeedEntry, FeedError} from "../feeds/FeedReader";
+import LogWrapper from "../LogWrapper";
+import { GetConnectionsResponseItem } from "../provisioning/api";
+import { IBridgeStorageProvider } from "../Stores/StorageProvider";
+import { BaseConnection } from "./BaseConnection";
+import markdown from "markdown-it";
+
+const log = new LogWrapper("FeedConnection");
+const md = new markdown();
+
+export interface FeedConnectionState extends IConnectionState {
+    url: string;
+}
+
+export class FeedConnection extends BaseConnection implements IConnection {
+    static readonly CanonicalEventType = "uk.half-shot.matrix-hookshot.feed";
+    static readonly EventTypes = [ FeedConnection.CanonicalEventType ];
+    private hasError = false;
+
+    public get feedUrl(): string {
+        return this.state.url;
+    }
+
+    constructor(
+        roomId: string,
+        stateKey: string,
+        private state: FeedConnectionState,
+        private readonly config: BridgeConfigFeeds,
+        private readonly as: Appservice,
+        private readonly storage: IBridgeStorageProvider
+    ) {
+        super(roomId, stateKey, FeedConnection.CanonicalEventType)
+        log.info(`FeedConnection created for ${roomId}, ${JSON.stringify(state)}`);
+    }
+
+    public isInterestedInStateEvent(eventType: string, stateKey: string): boolean {
+        return false;
+    }
+
+    public async handleFeedEntry(entry: FeedEntry): Promise<void> {
+        this.hasError = false;
+        const message = `New post in ${entry.feed.title}: [${entry.title}](${entry.link})`
+        await this.as.botIntent.sendEvent(this.roomId, {
+            msgtype: 'm.notice',
```

Comment:
I would also be nice if you could receive notifications from feeds, if you select (i.e. configure this on a per-feed basis?)

Reply:
I'm not sure what you mean by that. Do you mean to subscribe to a particular feed as an individual? That's something you should be able to do already by just creating an RSS room for yourself.

Reply:
I want to subscribe to a feed and select whether hookshot posts as `m.notice` (no notification) or `m.text` (notification). Then I can subscribe to "important" feeds and be explicitly notified when it updates.

Reply:
Ah, I see – that's a good point.

I wonder if this should be a global hookshot feature, or at least implemented in a way that can be reused easily across different services. What do you think @Half-Shot?

Reply:
Built a prototype in https://github.com/matrix-org/matrix-hookshot/pull/316

Reply:
Happy for this to land across all connections in a different PR.